index.js

```diff
@@ -190,16 +228,38 @@ function redisStore(args) {
     options = options || {};
 
     var ttl = (options.ttl || options.ttl === 0) ? options.ttl : redisOptions.ttl;
+    var gzip = (options.gzip || options.gzip === false) ? options.gzip : redisOptions.gzip;
+    if (gzip === true) {
+      gzip = gzipDefault;
+    }
 
     connect(function(err, conn) {
       if (err) {
         return cb && cb(err);
       }
       var val = JSON.stringify(value) || '"undefined"';
-      if (ttl) {
-        conn.setex(key, ttl, val, handleResponse(conn, cb));
-      } else {
-        conn.set(key, val, handleResponse(conn, cb));
+
+      if (gzip) {
+        zlib.gzip(val, gzip, function (gzErr, gzVal) {
+          if (gzErr) {
+            return cb && cb(gzErr);
+          }
+
+          if (ttl) {
+            conn.setex(key, ttl, gzVal, handleResponse(conn, cb));
+          }
+          else {
+            conn.set(key, gzVal, handleResponse(conn, cb));
+          }
+        });
+      }
+      else {
+        if (ttl) {
+          conn.setex(key, ttl, val, handleResponse(conn, cb));
+        }
+        else {
+          conn.set(key, val, handleResponse(conn, cb));
```

Comment:
I was thinking of something like this directly inside the set:

```js
      function persist(err, value) {
        if (gzErr) {
          return cb && cb(gzErr);
        }
        if (ttl) {
            conn.setex(key, ttl, value, handleResponse(conn, cb));
          }
          else {
            conn.set(key, value, handleResponse(conn, cb));
          }
      }

      if (gzip) {
        zlib.gzip(val, gzip, persist);
      }
      else {
        persist(null, val);
      }
```

Not sure if it is best practice, but I find it more readable than having multiple set commands inside the main object

Wooh, this project needs a good cleanup :sun_with_face: